test/hotspot/gtest/runtime/test_os_bsd_memory_tagging.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2019, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "unittest.hpp"
+
+#ifdef __APPLE__
+#include <mach/mach_vm.h>
+#include <mach/vm_statistics.h>
+#include <sys/mman.h>
+#include <cstring>
+
+// Test that memory allocations on macOS are properly tagged with VM_MEMORY_JAVA
+// This validates the VM_MAKE_TAG(VM_MEMORY_JAVA) changes in os_bsd.cpp
+class BSDMemoryTaggingTest : public ::testing::Test {
```

Comment:
This test does not test new JVM runtime code.
I found there are tests which might test this code:
- test/hotspot/gtest/runtime/test_committed_virtualmemory.cpp
- test/hotspot/gtest//runtime/test_os.cpp

You need to check if they cover your changes. If not, you need to add tests to `test_os_bsd.cpp`.
Also there are tests for ZGC in `test/hotspot/gtest/gc/z`. You need to check if any of them cover your changes.

IMO, `is_memory_tagged_as_java` will be needed.